### PR TITLE
Add validation to Summernote and fix its ValueChangeEvent.

### PR DESCRIPTION
**Pull request by BenDol**
It will manually fire the ValueChangeEvent onPaste and onKeyDown since the value change event doesn't invoke.

If someone knows why perhaps there is a better fix, but for now this works fine.

### CODE REVIEW

**Comment by BenDol** [comment]
This change also has setValidationEnabled/isValidationEnabled to sync this patch https://github.com/gwtbootstrap3/gwtbootstrap3/pull/405

**Comment by sjardine** [comment]
Checks have failed.  Can you please fix?

File: src/main/java/org/gwtbootstrap3/extras/summernote/client/ui/base/mixin/SummernoteValidatorMixin.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+import org.gwtbootstrap3.extras.summernote.client.event.SummernoteOnBlurEvent;
+import org.gwtbootstrap3.extras.summernote.client.event.SummernoteOnBlurHandler;
+import org.gwtbootstrap3.extras.summernote.client.ui.base.SummernoteBase;
+
+/**
+ * Summernote validator mixin. Contains all of the validation logic.
+ * 
+ * @author Ben Dol
+ */
+public class SummernoteValidatorMixin implements HasValidators<String> {
```
Comment (sjardine) [inline src/main/java/org/gwtbootstrap3/extras/summernote/client/ui/base/mixin/SummernoteValidatorMixin.java]: Shouldn't we just override the DefaultValidatorMixin?  This is creating a bunch of duplicate code.

Comment (BenDol) [inline src/main/java/org/gwtbootstrap3/extras/summernote/client/ui/base/mixin/SummernoteValidatorMixin.java]: Oops I was mean't to do that, sorry we had 2 earthquakes like 40 minutes ago so I just started committing my changes haha.

**Comment by BenDol** [comment]
If you reinvoke a travis build it should build with the PR merged on the main repo.

**Comment by crevete** [comment]
@BenDol Same validation issue... on the third-party wrapper, please extend summernote to do validation stuff. The wrapper is not supposed to do this. I'm sorry

**Comment by BenDol** [comment]
This widget already half supported it since it uses TextArea, it just wasn't binding to the onBlur correctly.

**Comment by BenDol** [comment]
also ValueChangeEvent wasn't firing properly. All valid fixes that should be accepted IMHO they should be accepted.

**Comment by crevete** [comment]
The ```blue``` typo should be fixed, right.

The ValueChangeEvent should not be fired like that. All existing ValueChangeEvent are fired when the focus leaves widget. So the original implementation is correct.

And the rest (about validation stuff), to be discussed...

**Comment by BenDol** [comment]
No the ValueChangeEvent isn't being fired, hence why I made the modifications. Try it for yourself.

**Pull request closed without merging**